pkg/controller/perconaservermongodb/pbm.go

```diff
@@ -209,10 +254,21 @@ func (r *ReconcilePerconaServerMongoDB) reconcilePiTRConfig(ctx context.Context,
 		return nil
 	}
 
-	stgName, _, err := cr.Spec.Backup.MainStorage()
-	if err != nil {
-		// no storage found
-		return nil
+	var stgName string
+	if cr.CompareVersion("1.20.0") >= 0 {
+		stgName, _, err = cr.Spec.Backup.MainStorage()
+		if err != nil {
+			// no storage found
+			return nil
+		}
+
+	} else {
+		if len(cr.Spec.Backup.Storages) == 1 {
```

Comment:
Is this check really needed? Looping the storages will return immediately if len=0.

Reply:
not needed